content/en/continuous_integration/setup_tests/java.md

```diff
@@ -107,91 +97,62 @@ Configure the [Maven Surefire Plugin][3] and/or the [Maven Failsafe Plugin][4] t
 </plugin>
 {{< /code-block >}}
 
-Run your tests using the `ci-app` profile, for example:
+Run your tests as you normally do, specifying the environment where tests are being run (e.g. `local` when running tests on a developer workstation, or `ci` when running them on a CI provider) in the `DD_ENV` environment variable. For example:
 
 {{< code-block lang="bash" >}}
-mvn clean verify -P ci-app
+DD_ENV=ci mvn clean verify -P ci-app
 {{< /code-block >}}
 
 ### Gradle
 
-Configure the `test` Gradle task by adding to the `jvmArgs` attribute the `-javaagent` argument targeting the Datadog Java tracer based on the `configurations.ddTracerAgent` property (use `-Ddd.integration.testng.enabled=true` if your testing framework is TestNG rather than JUnit):
+Configure the `test` Gradle task by adding to the `jvmArgs` attribute the `-javaagent` argument targeting the Datadog Java tracer based on the `configurations.ddTracerAgent` property, specifiying the name of the service or library under test with the `-Ddd.service` property:
 
-{{< code-block lang="groovy" >}}
+{{< code-block lang="groovy" filename="build.gradle" >}}
 test {
-    jvmArgs = ["-javaagent:${configurations.ddTracerAgent.asPath}", "-Ddd.prioritization.type=ENSURE_TRACE", "-Ddd.jmxfetch.enabled=false", "-Ddd.integrations.enabled=false", "-Ddd.integration.junit.enabled=true"]
+    jvmArgs = ["-javaagent:${configurations.ddTracerAgent.asPath}", "-Ddd.service=my-java-app", "-Ddd.prioritization.type=ENSURE_TRACE", "-Ddd.jmxfetch.enabled=false", "-Ddd.integrations.enabled=false", "-Ddd.integration.junit.enabled=true", "-Ddd.integration.testng.enabled=true"]
 }
 {{< /code-block >}}
 
-Run your tests as you normally do, for example:
+Run your tests as you normally do, specifying the environment where test are being run (e.g. `local` when running tests on a developer workstation, or `ci` when running them on a CI provider) in the `DD_ENV` environment variable. For example:
 
 {{< code-block lang="bash" >}}
-./gradlew cleanTest test --rerun-tasks
+DD_ENV=ci ./gradlew cleanTest test --rerun-tasks
```

Comment:
Probably we should add here some property discriminant to be used to activate CI App. Now, we're activating it by default for Gradle. For Maven we're using profiles, though.

Reply:
Any suggestions to do that @drodriguezhdez ? It's the same problem as with JS and Ruby

Reply:
```suggestion
DD_ENV=ci ./gradlew cleanTest test --rerun-tasks -Pci-app
```

Using this approach, the command would be pretty similar to the Maven. 
Notice the suggestion in the code from above.